components/eip681/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+//! Parser for [EIP-681](https://eips.ethereum.org/EIPS/eip-681) transaction requests.
+//!
+//! The top level of the parsing tree is the type [`EthereumTransactionRequest`].
+//!
+//! ## ABNF Syntax
+//!
+//! ```abnf
+//! request          = schema_prefix target_address [ "@" chain_id ] [ "/" function_name ] [ "?" parameters ]
+//! schema_prefix    = "ethereum" ":" [ "pay-" ]
+//! target_address   = ethereum_address
+//! chain_id         = 1*DIGIT
+//! function_name    = STRING
+//! ethereum_address = ( "0x" 40*HEXDIG ) / ENS_NAME
+//! parameters       = parameter *( "&" parameter )
+//! parameter        = key "=" value
+//! key              = "value" / "gas" / "gasLimit" / "gasPrice" / TYPE
+//! value            = number / ethereum_address / STRING
+//! number           = [ "-" / "+" ] *DIGIT [ "." 1*DIGIT ] [ ( "e" / "E" ) [ 1*DIGIT ] ]
+//! ```
```

Comment:
It would be useful to include the definitions of `DIGIT`, `STRING`, `ENS_NAME`, and `TYPE`. For any of these that can't reasonably be defined precisely in ABNF, it's sufficient here to give a covering grammar and an informally stated restriction on it.